exporters/otlp/internal/retry/retry.go

```diff
@@ -15,6 +15,9 @@
 // Package retry provides request retry functionality that can perform
 // configurable exponential backoff for transient errors and honor any
 // explicit throttle responses received.
+//
+// Deprecated: package retry exists for historical compatibility, it should not
```

Comment:
I think we should also add the deprecation notice in `exporters/otlp/internal/retry/go.mod`